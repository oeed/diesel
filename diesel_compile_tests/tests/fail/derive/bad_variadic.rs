extern crate diesel;

use diesel::*;

#[declare_sql_function]
extern "SQL" {
    //~^ ERROR: invalid ABI: found `SQL`

    #[variadic(not_a_literal_number)]
    //~^ ERROR: expected integer literal, the correct format is `#[variadic(3)]`
    //~| ERROR: cannot find attribute `variadic` in this scope
    fn f();
}

#[declare_sql_function]
extern "SQL" {
    #[variadic(3)]
    //~^ ERROR: invalid variadic argument count: not enough function arguments
    fn g<A: SqlType>(a: A);
}

fn main() {}
