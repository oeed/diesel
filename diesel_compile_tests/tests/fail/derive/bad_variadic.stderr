error: expected integer literal, the correct format is `#[variadic(3)]`
 --> tests/fail/derive/bad_variadic.rs:9:16
  |
LL |     #[variadic(not_a_literal_number)]
  |                ^^^^^^^^^^^^^^^^^^^^

error: invalid variadic argument count: not enough function arguments
  --> tests/fail/derive/bad_variadic.rs:17:16
   |
LL |     #[variadic(3)]
   |                ^

error: cannot find attribute `variadic` in this scope
 --> tests/fail/derive/bad_variadic.rs:9:7
  |
LL |     #[variadic(not_a_literal_number)]
  |       ^^^^^^^^

error[E0703]: invalid ABI: found `SQL`
 --> tests/fail/derive/bad_variadic.rs:6:8
  |
LL | extern "SQL" {
  |        ^^^^^ invalid ABI
  |
  = note: invoke `rustc --print=calling-conventions` for a full list of supported calling conventions
For more information about this error, try `rustc --explain E0703`.
